### ISSUE TITLE
How to handle a start/skip/offset parameter instead of a page one?

### ISSUE THREAD

**Issue by chikamichi** [backbone-paginator/backbone.paginator#273]
Hi,

I am working with a paginated API endpoint which does not provide `currentPage` and `pageSize` properties. Instead, it works by providing `start` and `limit` properties, where `start` is an offset. For instance, the `{start: 51, limit: 25}` would return the results from the third page.

How could I handle this with backbone.paginator ? In  `parseState`, I can compute `pageSize` and `currentPage` from the provided properties (they map to `limit` and `Math.ceil(start/limit)` respectively), but when fetching from the server (say, when calling `get*Page`), I would need to be able to convert them back so as to provide the custom params to the server.

I cannot "hack" `currentPage` and set it be just `start` (the offset), because backbone.paginator's `getPage` assumes (with good reasons) that `currentPage` changes by unitary increments.

So, basically: how could I tweak backbone.paginator to handle an resource offset instead of a page number?

**Comment by chikamichi** [comment]
Somehow I managed to miss the part about extra query string parameters twice while reading the documentation :)

I got around my issue like this:

```coffeescript
state:
  firstPage: 0
  pageSize: 50

queryParams:
  currentPage: null
  start: ->
    @state.currentPage * @state.pageSize
  pageSize: 'limit'
  totalPages: null
  totalRecords: null

parseState: (result, queryParams, state, options) ->
  pageSize: result.limit
  currentPage: Math.ceil(result.start/result.limit)
```

**Comment by timwis** [comment]
Struggled to find this - were the docs removed? Thanks for posting the solution @chikamichi !
